### PR TITLE
Features/people picker

### PR DESCRIPTION
**Pull request by agtenr**
## Type of change

- [ ] Bugfix
- [ ] Component improvement
- [x] New component
- [ ] New package
- [ ] Other

## Description
This PR contains a new component: PeoplePicker:
 - Reusable People Picker which accepts users or groups
 - Extended from Fluent UI Base Picker
 - Mock provider is added for testing purposes
 - Storybook stories are added for basic usage and custom suggestion rendering

### CODE REVIEW

File: packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+/**
+ * Possible variations of group types in the Microsoft ecosystem.
+ */
+export enum GroupType {
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts:5]: Should we make this a flagged enum? Might need to support combination of choices.

File: packages/dw-react-controls/src/components/PeoplePicker/models/SearchType.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Possible object options to search for in the people picker
+ */
+export enum SearchType {
+	/**
+	 * Search for users and groups
+	 */
+	UsersAndGroups = "UsersAndGroups",
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/models/SearchType.ts:8]: Should this be explicit? Or better combined with flagged enum?

File: packages/dw-react-controls/src/components/PeoplePicker/stories/PeoplePicker.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import * as stories from "./PeoplePicker.stories";
+
+<Meta
+	title="Components/PeoplePicker"
+	component={PeoplePicker}
+/>
+
+# PeoplePicker
+
+The `PeoplePicker` controls allows you to select users and groups.
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/stories/PeoplePicker.stories.mdx:12]: Typo
```suggestion
The `PeoplePicker` control allows you to select users and groups.
```

File: stories/Contribution.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import { Meta, Link } from "@storybook/addon-docs";
```
Comment (NickSevens) [inline stories/Contribution.stories.mdx:1]: Rename to Contribut**ing**.stories.mdx

**Review by NickSevens — CHANGES_REQUESTED** [review]

File: packages/dw-react-controls/src/components/PeoplePicker/UserOrGroupPicker/UserOrGroupPicker.styles.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { getStyles as getBasePickerStyles } from "office-ui-fabric-react/lib/components/pickers/BasePicker.styles";
+import { concatStyleSets } from "office-ui-fabric-react/lib/Styling";
+import { IUserOrGroupPickerStyleProps, IUserOrGroupPickerStyles } from "./UserOrGroupPicker.types";
+
+export const getStyles = (props: IUserOrGroupPickerStyleProps): IUserOrGroupPickerStyles => {
+	const { styles: customStyles } = props;
+
+	const basePickerStyles: IUserOrGroupPickerStyles = getBasePickerStyles(props);
+
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/UserOrGroupPicker/UserOrGroupPicker.styles.ts:9]: Add error state to control (red border, etc).
See TermPicker.styles.ts for reference.

**Review by NickSevens — CHANGES_REQUESTED** [review]

File: packages/dw-react-controls/src/components/PeoplePicker/PeoplePicker.base.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+		currentSelection: (IUser | IGroup)[]
+	): Promise<(IUser | IGroup)[]> => {
+
+		if (!filter || filter.length === 0) {
+			return [];
+		}
+
+		const filterOptions: IPeoplePickerFilterOptions = {
+			idsToIgnore: currentSelection.map(i => i.id),
+			searchFor: searchFor ?? ObjectType.UsersOnly,
```
[the remark this answers is not in the document]
Comment (NickSevens) [reply packages/dw-react-controls/src/components/PeoplePicker/PeoplePicker.base.tsx]: @agtenr maybe we can make this a required property, so there's always an explicit choice.

[the remark this answers is not in the document]
Comment (NickSevens) [reply packages/dw-react-controls/src/components/PeoplePicker/PeoplePicker.base.tsx]: Maybe we should make this a required property, so the choice is always specific?

File: packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts (hunk 1 above)
Comment (agtenr) [reply packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts:5]: You can now provide it as an array (e.g. [GroupType.AAD, GroupType.M365]). 
Maybe it is indeed a bit more elegant to use the flagged enum

File: packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+	/**
+	 * Azure Active Directory Group
+	 */
+	AAD = 1 << 0,
+	/**
+	 * Microsoft 365 Group
+	 */
+  M365 = 1 << 1,
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts:14]:
```suggestion
	M365 = 1 << 1,
```

File: packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+	/**
+	 * SharePoint Online Security Group
+	 */
+  SPO = 1 << 2
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/models/GroupType.ts:18]:
```suggestion
	SPO = 1 << 2
```

File: packages/dw-react-controls/src/components/PeoplePicker/models/SearchType.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+ */
+export enum SearchType {
+	/**
+	 * Search for users
+	 */
+	Users = 1 << 0,
+	/**
+	 * Search for groups
+	 */
+	Groups = 1 << 1,
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/models/SearchType.ts:14]:
```suggestion
	Groups = 1 << 1,
	/**
	 * Search for any supported type
	 */
	Any = Users | Groups
```

File: packages/dw-react-controls/src/components/PeoplePicker/stories/PeoplePicker.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+Basic.argTypes = {
+	...defaultArgTypes,
+	searchFor: {
+		options: [1, 2, 3],
+		control: {
+			type: "radio",
+			labels: {
+					1: "Users",
+					2: "Groups",
+					3: "Users and groups"
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/PeoplePicker/stories/PeoplePicker.stories.tsx:55]:
```suggestion
					3: "Any"
```

**Review by NickSevens — APPROVED** [review]

**Review by NickSevens — COMMENTED** [review]
@agtenr as discussed: I removed the `SearchType` and `GroupType` from the component, and it should now be the responsibility of the `provider` instead.
Also, I've changed the type of the _selected items_ to a generic `PeoplePickerValue` type which contains less properties than an `IUser` or `IGroup`. This encourages a user to save less data to a DB for example, while still ensuring all required properties to show the current value are provided.

**Pull request merged**